tutorials/chapter2.md

```diff
@@ -274,6 +274,105 @@ int main(int argc, char **argv) {
 * The Environment class interface can be used to get and set values. When getting/setting a parameter, you must know the type of the parameter and use the relevant getter/setter. 
 * After running this example, you can verify that a new parameter `/paramFloat` exists with a value 1.5 using the `rosparam` command-line utility.
 
+#### Example 5: Exchange ROS Messages
+
+This example shows us that with Klepsydra core you can use ROS messages to exchange data between the nodes.
+
+```cpp
+#include "kpsr_tutorial_chp2_ros/simple_publisher.h"
+#include <iostream>
+#include <klepsydra/high_performance/event_loop_middleware_provider.h>
+#include <klepsydra/serialization/identity_mapper.h>
+#include <kpsr_ros_core/from_ros_middleware_provider.h>
+#include <kpsr_ros_core/to_ros_middleware_provider.h>
+#include <std_msgs/String.h>
+
+void callbackFunction(const std_msgs::String &message)
+{
+    std::cout << "Message received: " << message.data << std::endl;
+    std::cout << "Callback (ros subscriber) thread ID: " << std::this_thread::get_id() << std::endl;
+}
+
+void rosTosKpsr(ros::NodeHandle &nodeHandle)
+{
+    std::string topicName1 = "tutorial_api_example5_1";
+    ros::Publisher strPublisher = nodeHandle.advertise<std_msgs::String>(topicName1, 20);
+    kpsr::ros_mdlw::FromRosMiddlewareProvider fromRosProvider(nodeHandle);
+    // Declare the eventloop and vector publisher
+    kpsr::high_performance::EventLoopMiddlewareProvider<16> eventloop(nullptr);
+
+    eventloop.start();
+
+    // auto strpublisher = eventloop.getPublisher<std_msgs::String>("example5", 0, nullptr, nullptr);
```

Comment:
in fact, the idea is to use kpsrPublisher to publish data. This way, any class in the rest of your code that uses Publisher* can still work and will in fact use the underlying ros publisher to actually transmit data.

For example, the SimplePublisher class in these tutorial examples has a Publisher pointer. So you could use the SimplePublisher class with these ros hooks and now without changing your original class you have added ROS support. the only thing that changes is your calls in the main() and your original SimplePublisher is agnostic to ros.